src/schema/definition.rs

```diff
@@ -1,11 +1,85 @@
 //! Contains the full definition of a data system which Ardite will use.
 
+use linear_map::LinearMap;
 use schema::Schema;
+use value::Key;
 
 /// The definition object which contains all necessary information to
 /// understand an Ardite Schema Definition.
 #[derive(PartialEq, Debug)]
 pub struct Definition {
-  /// The shape of the defined data.
-  pub data: Schema
+  /// Types defined in the database.
+  pub types: LinearMap<Key, Type>
+}
+
+/// Represents a high-level database type.
+#[derive(PartialEq, Debug)]
+pub struct Type {
+  /// The schema used to validate data which claims to be of this type.
+  pub schema: Schema
+}
+
+/// A function to be used by tests which creates a complete basic definition.
+/// Should define the same schema which exists in the
+/// `tests/fixtures/definitions/basic.json` file.
+#[cfg(test)]
+pub fn create_basic() -> Definition {
+  use regex::Regex;
+  use schema::*;
+
+  Definition {
+    types: linear_map! {
+      S!("person") => Type {
```

Comment:
The `S!` macro? The point is that this macro has a short name as it does the super common (in testing) operation of converting `str` to `String`. This is the shortest and most readable alternative to `String::from("person")` or `"person".to_string()`.